redash/handlers/queries.py

```diff
@@ -350,4 +350,16 @@ def post(self, query_id):
 
 class QueryTagsResource(BaseResource):
     def get(self):
-        return {t[0]: t[1] for t in models.Query.all_tags(self.current_user, True)}
+        """
+        Returns all query tags including those for drafts.
+        """
+        tags = models.Query.all_tags(self.current_user, include_drafts=True)
+        return {
+            'tags': [
```

Comment:
Maybe it makes sense to change the model functions (`all_tags`) to return a dictionary object in the first place?

Reply:
I thought about that as well, but I tend to lean towards data model classes that return queries instead of serialized versions of the queries and do the transformation on the view layer depending on the needs of the view. That way the separation of concerns is kept and prevents the creation of "fat models", which deal with all kinds of app logic.

Would compacting this code help you?

```python
tags = models.Query.all_tags(self.current_user, include_drafts=True)
return {
    'tags': [dict(zip(['name', 'count'], tag)) for tag in tags]
}
```